src/system/file_path.cc

```diff
@@ -40,7 +40,7 @@ FilePath::FilePath(string file_path, const  FileType ft)
     if (ft == input_file) {
     	if ( FilePath::is_absolute_path(abs_file_path_) ) {
     	} else {
-            abs_file_path_ = root_dir + DIR_DELIMITER + abs_file_path_;
+            abs_file_path_ = input_root_dir + DIR_DELIMITER + abs_file_path_;
```

Comment:
Does boost have a portable way of path joining? I.e. could we use boost to get rid of the DIR_DELIMITER, it is set constantly to "/" anyway. 
Even if we keep current approach we should move it from system into class FilePath and possibly make it private (in future) there are few places where it is used out of FilePath and some of them are not too correct, e.g. the match method of Input::Type::FilePath.